### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite.

### ISSUE THREAD

**Issue by peterpeterparker** [jakearchibald/idb-keyval#128]
### Issue

After cloning the repo, I get following error while running `npm run dev`:

> rollup v2.56.3
> bundles test/index.ts → dist/test/index.js...
> test/index.ts:234:20 - error TS2589: Type instantiation is excessively deep and possibly infinite.
> 
> 234         typeAssert<IsExact<typeof result, IDBValidKey[]>>(true);
>                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> test/index.ts:315:20 - error TS2589: Type instantiation is excessively deep and possibly infinite.
> 
> 315         typeAssert<IsExact<typeof result, [IDBValidKey, any][]>>(true);
>                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> 
> Found 2 errors.
> 
> 
> /Users/.../projects/lab/idb-keyval/rollup.config.js:91
>           file: 'dist/esm/index.js',
>                 ^
> Error: TypeScript build failed
>     at ChildProcess.<anonymous> (/Users/.../projects/lab/idb-keyval/rollup.config.js:91:17)
>     at ChildProcess.emit (node:events:394:28)
>     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)

### Environment

Node v16.5.0
Npm 7.19.1
OSX

### Replicate

```
git clone https://github.com/jakearchibald/idb-keyval/
cd idb-keyval
npm ci
npm run dev
```

**Comment by jakearchibald** [comment]
Ah, I encountered this with another library. I probably need to downgrade typescript, but also file a bug about the underlying issue.

**Comment by jakearchibald** [comment]
Fixed in 1e0eb5c (by downgrading typescript). Seems to be a TypeScript bug https://github.com/microsoft/TypeScript/issues/45672
